shared/src/commonMain/kotlin/org/hyperskill/app/track_list/view/TrackListViewStateMapper.kt

```diff
@@ -0,0 +1,45 @@
+package org.hyperskill.app.track_list.view
+
+import kotlin.math.floor
+import org.hyperskill.app.SharedResources
+import org.hyperskill.app.core.view.mapper.ResourceProvider
+import org.hyperskill.app.track.domain.model.TrackWithProgress
+import org.hyperskill.app.track_list.presentation.TrackListFeature
+
+class TrackListViewStateMapper(
+    private val resourceProvider: ResourceProvider
+) {
+    fun map(state: TrackListFeature.State): TrackListViewState =
+        when (state) {
+            TrackListFeature.State.Idle -> TrackListViewState.Idle
+            TrackListFeature.State.Loading -> TrackListViewState.Loading
+            TrackListFeature.State.NetworkError -> TrackListViewState.Error
+            is TrackListFeature.State.Content -> TrackListViewState.Content(
+                selectedTrack = state.tracksWithProgresses.firstOrNull {
+                    it.track.id == state.selectedTrackId
+                }?.let(::mapTrackToViewStateTrack),
+                tracks = state.tracksWithProgresses
+                    .filter { it.track.id != state.selectedTrackId }
+                    .sortedBy { it.trackProgress.rank }
+                    .map(::mapTrackToViewStateTrack)
+            )
+        }
+
+    private fun mapTrackToViewStateTrack(trackWithProgress: TrackWithProgress): TrackListViewState.Track =
+        TrackListViewState.Track(
+            id = trackWithProgress.track.id,
+            imageSource = trackWithProgress.track.cover ?: "",
+            title = trackWithProgress.track.title,
+            description = trackWithProgress.track.description,
+            timeToComplete = getTimeToComplete(trackWithProgress.track.secondsToComplete),
+            rating = trackWithProgress.trackProgress.averageRating.toString(),
```

Comment:
`rating` should be formatted either:

1. When `averageRating` exists -> should be formatted with one decimal point `4.7122` -> `4.7` (see for example iOS [Formatter.averageRating](https://github.com/hyperskill/mobile-app/blob/develop/iosHyperskillApp/iosHyperskillApp/Sources/Helpers/Formatter.swift#L16))
2. When `averageRating` doesn't exists -> `New` (comment in the [YouTrack](https://vyahhi.myjetbrains.com/youtrack/issue/ALTAPPS-750/Shared-Create-TrackListFeature#focus=Comments-74-232072.0-0))